src/Temporalio/Workflows/WorkflowDefinition.cs

```diff
@@ -286,6 +304,27 @@ public static WorkflowDefinition Create(
                         runMethod = method;
                     }
                 }
+                if (method.IsDefined(typeof(WorkflowDynamicOptionsAttribute), false))
```

Comment:
Did we decide whether any workflow can define this or if only dynamic workflows can? If any, I wonder if the term "dynamic" is confusing. I wonder if `[WorkflowOptionsProvider]` or something is a better name.

Reply:
It's only dynamic workflows. Added enforcement.